tests/test_appliancesmanager.py

```diff
@@ -1,123 +1,148 @@
-        )
 
-    # this should always be called
-    assert_appliances_manager_call(
-        http_client_mock,
-        get_appliances_idx,
-        BACKEND_SELECTOR_MOCK.get_owned_appliances_url(ACCOUNT_ID),
-        forbidden_headers=["WP-CLIENT-BRAND"],
-    )
+async def test_user_details_called_if_account_id_not_known(
+    backend_selector_mock: BackendSelector,
+    appliances_manager: AppliancesManager,
+    aioresponses_mock,
+):
+    await appliances_manager.fetch_appliances()
 
-    # this should always be called and requires the WP-CLIENT-BRAND header
-    assert_appliances_manager_call(
-        http_client_mock,
-        get_appliances_idx + 1,
-        BACKEND_SELECTOR_MOCK.shared_appliances_url,
-        {"WP-CLIENT-BRAND": "DUMMY_BRAND"},
+    aioresponses_mock.assert_any_call(
+        URL(backend_selector_mock.user_details_url), "GET"
     )
 
-    # ensure that the washer_dryers list is populated
-    assert len(am.washer_dryers) == 2
 
-    await http_client_mock.close_session()
+async def test_get_owned_appliances_call_does_not_contain_wp_client_brand_header(
+    backend_selector_mock: BackendSelector,
+    appliances_manager: AppliancesManager,
+    aioresponses_mock,
+):
+    await appliances_manager.fetch_appliances()
+
+    # call to get owned appliances - this one cannot contain the WP-CLIENT-BRAND header
+    owned_call = aioresponses_mock.requests[
+        ("GET", URL(backend_selector_mock.get_owned_appliances_url(ACCOUNT_ID)))
+    ]
+    assert len(owned_call) == 1
+    owned_call_headers = owned_call[0][1]["headers"]
+    assert "WP-CLIENT-BRAND" not in owned_call_headers
+
+
+async def test_get_shared_appliances_call_contains_wp_client_brand_header(
+    backend_selector_mock: BackendSelector,
+    appliances_manager: AppliancesManager,
+    aioresponses_mock,
+):
+    await appliances_manager.fetch_appliances()
+
+    # call to get shared appliances - this one should contain the WP-CLIENT-BRAND header
+    shared_call = aioresponses_mock.requests[
+        ("GET", URL(backend_selector_mock.shared_appliances_url))
+    ]
+    assert len(shared_call) == 1
+    shared_call_headers = shared_call[0][1]["headers"]
+    assert shared_call_headers["WP-CLIENT-BRAND"] == "DUMMY_BRAND"
+
+
+async def test_fetch_appliances_returns_owned_and_shared_appliances(
+    appliances_manager: AppliancesManager,
+):
+    await appliances_manager.fetch_appliances()
+
+    # the test data has one owned appliance and one shared appliance
+    # so if this is 2 then we have both
+    assert len(appliances_manager.washer_dryers) == 2
+
+
+async def test_fetch_appliances_calls_owned_and_shared_methods(
+    appliances_manager: AppliancesManager,
+):
+    shared_appliance_mock = get_mock_coro(None)
+    owned_appliance_mock = get_mock_coro(None)
+
+    appliances_manager._get_shared_appliances = shared_appliance_mock
+    appliances_manager._get_owned_appliances = owned_appliance_mock
```

Comment:
Can we do this check using the http requests instead of altering internal behavior of the class? We could then probably also merge it `test_fetch_appliances_returns_owned_and_shared_appliances`

Reply:
Yep, that's a good idea, I've implemented it